src/common/ServerPushSetupBanner.js

```diff
@@ -0,0 +1,88 @@
+/* @flow strict-local */
+
+import React from 'react';
+import type { Node } from 'react';
+import isAfter from 'date-fns/isAfter';
+import subWeeks from 'date-fns/subWeeks';
+
+import ZulipBanner from './ZulipBanner';
+import { useSelector, useGlobalSelector, useDispatch } from '../react-redux';
+import { getIdentity, getAccount } from '../account/accountsSelectors';
+import { getIsAdmin, getRealm, getGlobalSettings } from '../directSelectors';
+import { dismissServerPushSetupNotice } from '../account/accountActions';
+import { openLinkWithUserPreference } from '../utils/openLink';
+
+type Props = $ReadOnly<{||}>;
+
+/**
+ * A "nag banner" saying the server hasn't enabled push notifications, if so
+ *
+ * When dismissed, reappears again in two weeks.
+ */
+export default function PushNotifsSetupBanner(props: Props): Node {
+  const dispatch = useDispatch();
+  const lastDismissedServerPushSetupNotice = useSelector(
+    state => getAccount(state).lastDismissedServerPushSetupNotice,
+  );
+  const pushNotificationsEnabled = useSelector(state => getRealm(state).pushNotificationsEnabled);
+  const realm = useSelector(state => getIdentity(state).realm);
+  const isAdmin = useSelector(getIsAdmin);
+  const settings = useGlobalSelector(getGlobalSettings);
+
+  let visible = false;
+  let text = '';
+  if (pushNotificationsEnabled) {
+    // don't show
+  } else if (
+    lastDismissedServerPushSetupNotice !== null
+    && isAfter(
+      lastDismissedServerPushSetupNotice,
+      subWeeks(
+        // TODO: Could rerender this component on an interval, to give an
+        //   upper bound on how old this "current date" can be.
+        new Date(),
+        2,
+      ),
+    )
+  ) {
+    // don't show
+  } else {
+    visible = true;
+    text = isAdmin
+      ? {
+          text:
+            'The Zulip server at {realm} is not set up to deliver push notifications. Please contact your administrator.',
+          values: { realm: realm.toString() },
+        }
+      : {
+          text: 'The Zulip server at {realm} is not set up to deliver push notifications.',
+          values: { realm: realm.toString() },
+        };
+  }
+
+  return (
+    <ZulipBanner
+      visible={visible}
+      text={text}
+      buttons={[
+        {
+          id: 'dismiss',
+          label: isAdmin ? 'Fix later' : 'Remind me later',
+          onPress: () => {
+            dispatch(dismissServerPushSetupNotice());
+          },
+        },
+        {
+          id: 'resolve',
+          label: isAdmin ? 'Fix now' : 'Learn more',
```

Comment:
"Fix now" doesn't feel right to me here -- as if it's a button that *will* fix it, rather than give you instructions on how you can fix it. (And it may not even be that, if you're an admin within Zulip but not a server admin with access to the underlying machine -- then it's instructions you can send a colleague in asking them to fix it.)

I think the same "Learn more" / "Remind me later" text as for non-admins might work best.